meilisearch/src/routes/metrics.rs

```diff
@@ -0,0 +1,45 @@
+use actix_web::http::header;
+use actix_web::web::{self, Data};
+use actix_web::HttpResponse;
+use index_scheduler::IndexScheduler;
+use meilisearch_auth::{AuthController, AuthFilter};
+use meilisearch_types::error::ResponseError;
+use meilisearch_types::keys::actions;
+use prometheus::{Encoder, TextEncoder};
+
+use crate::extractors::authentication::policies::ActionPolicy;
+use crate::extractors::authentication::GuardedData;
+use crate::routes::create_all_stats;
+
+pub fn configure(config: &mut web::ServiceConfig) {
+    config.service(web::resource("").route(web::get().to(get_metrics)));
+}
+
+pub async fn get_metrics(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::METRICS_GET }>, Data<IndexScheduler>>,
+    auth_controller: GuardedData<ActionPolicy<{ actions::METRICS_GET }>, AuthController>,
+) -> Result<HttpResponse, ResponseError> {
+    let response = create_all_stats(
+        (*index_scheduler).clone(),
+        (*auth_controller).clone(),
+        // we don't use the filters contained in the `ActionPolicy` because the metrics must have the right to access all the indexes.
+        &AuthFilter::default(),
```

Comment:
1. Thanks for the docs comment 🙂 
2. I think this has the potential to be surprising to users. I think that in no circumstances should the existence of an index not authorized by an API key be revealed. 
3. If we're going to go this route, maybe we could use `AuthFilter::with_allowed_indexes` instead to at least not have the `allow_index_creation` boolean set to true?

```suggestion
        &AuthFilter::with_allowed_indexes(std::iter::once("*").collect()),
```

Reply:
> I think that in no circumstances should the existence of an index not authorized by an API key be revealed.

Oh, but it'll be revealed whatever happens since there is a middleware telling you which routes were called and how many times 😬 
The only information you'll lose from what I see is the size of the non-authorized indexes.

-----

I think this discussion could be interesting for you @gmourier.
FYI, currently we wrote [Provides access to `GET` `/metrics` route. **A restriction on `indexes` does not affect returned metrics.**](https://github.com/meilisearch/specifications/pull/174/files#diff-efbeaa05fe855150edfa47dc218874a3d03dda4000c79375238288abd551694aR331) but that can change.

Reply:
> Oh, but it'll be revealed whatever happens since there is a middleware telling you which routes were called and how many times 😬

I see.

A maybe more specific issue I can see is that if an API key is restricted on indexes but not actions, one can use the metrics route to get information on which indexes are in existence.

Reply:
Another possibility I'm thinking of is to reject access to the metrics route with an error if the corresponding API key doesn't have `filters().all_indexes_authorized()`, so that a user must explicitly have all indexes authorized + the metrics action to access the metrics route.

Reply:
Ah, yes, that could be a nice solution indeed!

Reply:
Since it's experimental, could we keep it as it is and note it for a future iteration? (I've planned to create a dedicated production discussion for the whole Experimental /metrics)

I sense `/metrics` will mostly be consumed with an API Key having access to all indexes initially because the user first use case is to monitor everything happening on the instance.

Reply:
discussed with @gmourier and @irevoire: we'll be switching to the following:

>  Reject access to the metrics route with an error if the corresponding API key doesn't have filters().all_indexes_authorized(), so that a user must explicitly have all indexes authorized + the metrics action to access the metrics route.

Reply:
Thanks, @dureuill!